helix-dap/src/client.rs

```diff
@@ -29,10 +29,12 @@ pub struct Client {
     pub caps: Option<DebuggerCapabilities>,
     //
     pub breakpoints: HashMap<PathBuf, Vec<SourceBreakpoint>>,
-    // TODO: multiple threads support
-    pub stack_pointer: Option<StackFrame>,
+    // thread_id -> frames
+    pub stack_frames: HashMap<usize, Vec<StackFrame>>,
```

Comment:
Using vector here allows up to go back. TODO: do it!

Reply:
Definitely, dap-mode has both "sf" which opens a picker as well as up/down to move over the stack. Should be as easy as changing the active frame

Reply:
Yeah. I'm gonna integrate vscode-cpptools (need a user preference for this!) and so make use of GDB and RR debuggers. RR receives `eval` commands, we probably wanna add shortcuts for them as well. I do hope rr works fine on Zen 2.